presto-main/src/main/java/com/facebook/presto/cost/JoinStatsRule.java

```diff
@@ -158,9 +161,14 @@ private PlanNodeStatsEstimate computeInnerJoinStats(JoinNode node, PlanNodeStats
         }
 
         PlanNodeStatsEstimate equiJoinEstimate = filterByEquiJoinClauses(crossJoinStats, node.getCriteria(), session, types);
-
         if (equiJoinEstimate.isOutputRowCountUnknown()) {
-            return PlanNodeStatsEstimate.unknown();
+            double defaultJoinSelectivityFactor = getDefaultJoinSelectivityCoefficient(session);
+            if (defaultJoinSelectivityFactor == DEFAULT_JOIN_SELECTIVITY_DISABLED) {
```

Comment:
Best not to use `==` comparison here and use [other methods](https://www.baeldung.com/java-comparing-doubles)

Reply:
good catch, change to `compare` method for accurately comparing two primitive double type variables